src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -108,138 +170,243 @@ public override void Initialize(AnalysisContext context)
                     OperationKind.Unary,
                     OperationKind.Binary,
                     OperationKind.ArrayCreation,
-                    OperationKind.CatchClause
+                    OperationKind.CatchClause,
+                    OperationKind.TypeOf,
+                    OperationKind.EventAssignment
                     );
 
-                // Handle library side definitions of preview features
-                context.RegisterSymbolAction(context => AnalyzeSymbol(context, requiresPreviewFeaturesSymbols, requiresPreviewFeaturesSymbolsToUsageType, virtualStaticsInInterfaces, previewFeaturesAttribute), s_symbols);
+                // Handle preview symbol definitions
+                context.RegisterSymbolAction(context => AnalyzeSymbol(context, requiresPreviewFeaturesSymbols, virtualStaticsInInterfaces, previewFeaturesAttribute), s_symbols);
             });
         }
 
-        private static bool ProcessTypeSymbolAttributes(ITypeSymbol symbol,
+        private static void ProcessFieldSymbolAttributes(SymbolAnalysisContext context, IFieldSymbol symbol,
+            ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols,
+            INamedTypeSymbol previewFeatureAttributeSymbol)
+        {
+            ISymbol symbolType = symbol.Type;
+            if (symbolType is IArrayTypeSymbol arrayType)
+            {
+                symbolType = arrayType.ElementType;
+            }
```

Comment:
Do we need to iteratively apply this until the symbolType is no longer array symbol?

Reply:
Nice catch. Fixed

Reply:
Ok fixed this locally. Will push it up in the next commit.